File: cvbrk/src/main/java/ai/analys/cvbrk/entity/Cv.java
package ai.analys.cvbrk.entity;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.hibernate.annotations.CreationTimestamp;

import java.time.LocalDateTime;
import java.util.List;

@Entity
@Builder
@AllArgsConstructor
@NoArgsConstructor
@Data
public class Cv {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String titre;
    private String path;


    @CreationTimestamp
    @Column(nullable = false, updatable = false)
    private LocalDateTime creatAt;

    @ManyToOne
    @JoinColumn(name = "etudiant_id", nullable = false)
    private User etudiant;

    @OneToMany(mappedBy = "etudiant", cascade = CascadeType.ALL)
    private List<Postule> postulants;


}

File: cvanalyser/dto/Cv.py
# dto/Cv.py
from dataclasses import dataclass

@dataclass
class Cv:
    id: int
    titre: str
    path: str
    creat_at: str

def parse_cv(data):
    return Cv(
        id=data["id"],
        titre=data["titre"],
        path=data["path"],
        creat_at=data["creat_at"]
    )